lib/components/features/lista.dart

```diff
@@ -1,28 +1,135 @@
+import 'package:bytebank/components/progresso.dart';
+import 'package:bytebank/models/feature.dart';
+import 'package:bytebank/webclient/features_webclient.dart';
 import 'package:flutter/material.dart';
 
 import 'item.dart';
 
-class ListaFeatures extends StatelessWidget {
+const _featureHistorico = 'historico';
+const _tituloFeatureHistorico = 'Histórico';
+
+const _featureTransferir = 'transferir';
+const _tituloFeatureTransferir = "Transferir";
+
+const _featureCartaoDeCredito = 'cartao de credito';
+const _tituloFeatureCartaoDeCredito = 'Cartão de crédito';
+
+const _featureAjuda = 'ajuda';
+const _tituloFeatureAjuda = 'Ajuda';
+
+class ListaFeatures extends StatefulWidget {
+  final Function() quandoClicaTransacoes;
+  final Function() quandoClicaContatos;
+
+  const ListaFeatures({
+    this.quandoClicaTransacoes,
+    this.quandoClicaContatos,
+  });
+
+  @override
+  State<StatefulWidget> createState() {
+    return _ListaFeaturesState();
+  }
+}
+
+class _ListaFeaturesState extends State<ListaFeatures> {
+  List<ItemFeature> _features = List();
+
+  @override
+  void initState() {
+    super.initState();
+    _buscaTodas();
+  }
+
   @override
   Widget build(BuildContext context) {
     return Padding(
       padding: const EdgeInsets.all(8.0),
       child: Container(
         height: 120,
-        child: ListView(
-          scrollDirection: Axis.horizontal,
-          children: <Widget>[
-            FeatureItem(
-              icone: Icons.monetization_on,
-              texto: 'Transferência',
-            ),
-            FeatureItem(
-              icone: Icons.people,
-              texto: 'Contatos',
-            ),
-          ],
-        ),
+        child: _tentaCarregarFeatures(),
       ),
     );
   }
+
+  Widget _tentaCarregarFeatures() {
+    if (_features.isEmpty) {
+      return Progresso();
+    }
+    return Features(_features);
+  }
+
+  _buscaTodas() async {
+    final List<Feature> features = await FeatureWebClient().todas();
+    final itemFeaturesDisponiveis = _carregaFeaturesDisponiveis(features);
+
+    //TODO código apenas para carregar padrões features caso não tenha comunicação com API
+    if (itemFeaturesDisponiveis.isNotEmpty) {
+      itemFeaturesDisponiveis.add(_devolveItemFeature(
+          Feature(_featureTransferir, true)));
+      itemFeaturesDisponiveis.add(_devolveItemFeature(
+          Feature(_featureHistorico, true)));
+    }
+
+    //TODO estou usando essa técnica para atualizar a lista, existe alguma outra abordagem recomendada?
```

Comment:
Ok, then I'll keep this solution for update the UI, because the delegate pattern not will explaned for now.

Oh, sorry for `_features`, I had put it as final, idk because I back for mutable variable, considering the final variable, I keep using `addAll()` or have another suggetions?

And about the only a load, yep, this solution was done for to load when the Widget is showed. Exists other recomended way to do it?

Reply:
It was just confusing when I read it since `addAll` was there, I assumed this to be an ever-growing list of features.

About the loading once, this is it for stateful widgets, there are other lifecycle methods but they may be called multiple times according to the presentation stack and application state.

Reply:
This list does not grow hehe
So, can I keep this approach? Remembering that I want to load features just once.